base/namedtuple.jl

```diff
@@ -179,8 +179,11 @@ nextind(@nospecialize(t::NamedTuple), i::Integer) = Int(i)+1
 convert(::Type{NamedTuple{names,T}}, nt::NamedTuple{names,T}) where {names,T<:Tuple} = nt
 convert(::Type{NamedTuple{names}}, nt::NamedTuple{names}) where {names} = nt
 
-function convert(::Type{NamedTuple{names,T}}, nt::NamedTuple{names}) where {names,T<:Tuple}
-    NamedTuple{names,T}(T(nt))::NamedTuple{names,T}
+function convert(::Type{NamedTuple{names,T1}}, nt::NamedTuple{names,T2}) where {names,T1<:Tuple,T2<:Tuple}
+    if (T1<:T2)
+        return nt
+    end
+    NamedTuple{names,T1}(T1(nt))::NamedTuple{names,T1}
 end
```

Comment:
It is a little more complicated than that, since T1 is not defined. We can compute it though, with some effort:
```suggestion
convert(::Type{NT}, nt::NT) where {names, NT<:NamedTuple{names}} = nt
convert(::Type{NT}, nt::NT) where {names, T<:Tuple, NT<:NamedTuple{names,T}} = nt
function convert(::Type{NT}, nt::NamedTuple{names}) where
                       {names, T<:Tuple, NT<:NamedTuple{names,T}}
    if !@isdefined T
       # converting abstract NT to an abstract Tuple type, to a concrete NT1, is not straightforward, so this could just be an error, but we define it anyways
       # _tuple_error(NT, nt)
       T1 = Tuple{ntuple(i -> fieldtype(NT, i), Val(length(names)))...}
       NT1 = NamedTuple{names, T1}
    else
       T1 = T
       NT1 = NT
    end
    return NT1(T1(nt))::NT1::NT
end
```

various examples
```julia
julia> convert(NamedTuple{(:a,),Tuple{I}} where I, (;a=1))
(a = 1,)

julia> convert(NamedTuple{(:a,),Tuple{Any}}, (;a=1))
NamedTuple{(:a,), Tuple{Any}}((1,))

julia> convert(NamedTuple{(:a,),}, (;a=1))
(a = 1,)

julia> convert(NamedTuple{(:a,),Tuple{I}} where I<:Unsigned, (;a=1))
NamedTuple{(:a,), Tuple{Unsigned}}((0x0000000000000001,))

julia> convert(NamedTuple{(:a,),Tuple{I}} where I<:AbstractString, (;a=1))
ERROR: MethodError: Cannot `convert` an object of type Int64 to an object of type AbstractString

julia> convert(NamedTuple, (;a=1))
(a = 1,)
```